dev/view.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <memory>   //  std::shared_ptr
+#include <string>   //  std::string
+#include <utility>  //  std::forward, std::move
+#include <sqlite3.h>
+#include <system_error> //  std::system_error
+#include <tuple>    //  std::tuple, std::make_tuple
+
+#include "database_connection.h"
+#include "row_extractor.h"
+#include "statement_finalizer.h"
+#include "error_code.h"
+#include "iterator.h"
+
+namespace sqlite_orm {
+    
+    namespace internal {
+        
+        template<class T, class S, class ...Args>
+        struct view_t {
+            using mapped_type = T;
+            using storage_type = S;
+            using self = view_t<T, S, Args...>;
+            
+            storage_type &storage;
+            std::shared_ptr<internal::database_connection> connection;
+            std::tuple<Args...> args;
+            
+            view_t(storage_type &stor, decltype(connection) conn, Args&& ...args_):
+            storage(stor),
+            connection(std::move(conn)),
+            args(std::make_tuple(std::forward<Args>(args_)...)){}
+            
+            size_t size() {
+                return this->storage.template count<T>();
+            }
+            
+            bool empty() {
+                return !this->size();
+            }
+            
+            iterator_t<self> end() {
```

Comment:
I think it is good to provide right const `end` and `begin` function and also provide `cend` and `cbegin` function for const access to `view`